File: training/src/classesandobjects/java/Logger.java
package classesandobjects.java;

public class Logger {
    private static final Logger INSTANCE = new Logger();

    private Logger() {
    }

    public static Logger getLogger() {
        return INSTANCE;
    }

    public void log(String level, String string) {
        System.out.printf("%s %s%n", level, string);
    }

    public static void main(String[] args) {
        Logger.getLogger().log("INFO", "March, no voy a mentirte");
    }
}

File: training/src/classesandobjects/scala/Logger.scala
package classesandobjects.scala

object Logger {
  def log(level: String, string: String): Unit = {
    printf("%s %s%n", level, string)
  }

  def main(args: Array[String]): Unit = {
    Logger.log("INFO", "March, no voy a mentirte")
  }
}

